gui/src/main.rs

```diff
@@ -55,33 +62,44 @@ impl Default for GuiConfig {
                     String::from("no-cache"),
                 ),
             ])),
-            response: None,
+            import_window_open: false,
+            import_file_path: String::from(""),
+            import_mode: ImportMode::COLLECTION,
         }
     }
 }
 
 pub struct Gui {
-    pub config: GuiConfig,
-    pub rt: runtime::Runtime,
+    pub config: Arc<RwLock<GuiConfig>>,
+    pub response: Arc<RwLock<Option<Value>>>,
 }
 impl Default for Gui {
     fn default() -> Self {
         Self {
-            config: GuiConfig::default(),
-            rt: runtime::Builder::new_multi_thread()
-                .enable_all()
-                .build()
-                .unwrap(),
+            config: Arc::new(RwLock::new(GuiConfig::default())),
+            response: Arc::new(RwLock::new(None)),
         }
     }
 }
 impl Gui {
-    fn spawn_submit(&mut self, input: HttpRequest) -> Result<Value, Box<dyn Error>> {
-        let result = self.rt.block_on(async {
-            let api_response = PostieApi::make_request(input).await;
-            api_response
+    async fn submit(input: HttpRequest) -> Result<Value, Box<dyn Error>> {
+        PostieApi::make_request(input).await
+    }
+    fn spawn_submit(&self, input: HttpRequest) -> Result<(), Box<dyn Error>> {
+        let result_for_worker = self.response.clone();
+        tokio::spawn(async move {
+            match Gui::submit(input).await {
+                Ok(res) => {
+                    println!("Res: {}", res);
+                    let mut result_write_guard = result_for_worker.try_write().unwrap();
+                    *result_write_guard = Some(res);
```

Comment:
From inside the other thread, the * is the dereference operator. This is essentially updating the value of the memory in the main thread.